### PR TITLE
GH-52093 : AttributeError raised with pd.concat between a None and Timestamp

### LINKED ISSUE #52093: BUG: AttributeError raised with pd.concat between a None and Timestamp
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
data = [{'A':None}]
df = pd.DataFrame(data)
data_2 = [{'A': pd.to_datetime("1990-12-20 00:00:00+00:00")}]
df_2 = pd.DataFrame(data_2)
pd.concat([df, df_2])
```

### Issue Description

When running `pd.concat` between two DataFrame: one with `None` as an object, and a second, with a timezone-enabled `pandas._libs.tslibs.timestamps.Timestamp` an `AttributeError` is raised.

```
AttributeError: 'NoneType' object has no attribute 'value'
```

### Expected Behavior

I expect this to return the same response when a non-timezone-enabled `pandas._libs.tslibs.timestamps.Timestamp`, which is the `None` changes to a `pd.NaT`

### Installed Versions

INSTALLED VERSIONS
------------------
commit           : c2a7f1ae753737e589617ebaaff673070036d653
python           : 3.11.2.final.0
python-bits      : 64
OS               : Linux
OS-release       : 5.10.102.1-microsoft-standard-WSL2
Version          : #1 SMP Wed Mar 2 00:30:59 UTC 2022
machine          : x86_64
processor        :
byteorder        : little
LC_ALL           : None
LANG             : C.UTF-8
LOCALE           : en_US.UTF-8

pandas           : 2.0.0rc1
numpy            : 1.24.2
pytz             : 2022.7.1
dateutil         : 2.8.2
setuptools       : 65.5.1
pip              : 22.3.1
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : None
gcsfs            : None
matplotlib       : None
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : None
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
zstandard        : None
tzdata           : None
qtpy             : None
pyqt5            : None

### PR DESCRIPTION
**Pull request by ramvikrams**
- [x] closes #52093 
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/internals/test_internals.py [hunk 1]
```diff
@@ -1432,3 +1432,11 @@ def test_make_block_no_pandas_array(block_maker):
         )
         assert result.dtype.kind in ["i", "u"]
         assert result.is_extension is False
+
+
+def test_concat_value():
+    data = [{"A": None}]
```
Comment (mroeschke) [inline pandas/tests/internals/test_internals.py:1438]: Could you move this test to pandas/tests/reshape/concat/test_concat.py

and also add the Github issue number as a comment?

File: pandas/tests/internals/test_internals.py [hunk 2]
```diff
@@ -1432,3 +1432,11 @@ def test_make_block_no_pandas_array(block_maker):
         )
         assert result.dtype.kind in ["i", "u"]
         assert result.is_extension is False
+
+
+def test_concat_value():
```
Comment (mroeschke) [inline pandas/tests/internals/test_internals.py:1437]:
```suggestion
def test_concat_none_to_dttz():
```

File: pandas/tests/internals/test_internals.py [hunk 3]
```diff
@@ -1432,3 +1432,11 @@ def test_make_block_no_pandas_array(block_maker):
...
+    data = [{"A": None}]
+    df = DataFrame(data)
+    data_2 = [{"A": pd.to_datetime("1990-12-20 00:00:00+00:00")}]
+    df_2 = DataFrame(data_2)
+    pd.concat([df, df_2])
```
Comment (mroeschke) [inline pandas/tests/internals/test_internals.py:1442]:
```suggestion
    result = pd.concat([df, df_2])
```

And then construct the `expected = DataFrame(...)` result and then use `tm.assert_frame_equal(result, expected)`

**Review by mroeschke — COMMENTED** [review]
Could you add a whatsnew note in `v2.0.1.rst` too

File: doc/source/whatsnew/v2.0.1.rst [hunk 4]
```diff
@@ -20,10 +20,12 @@ Fixed regressions
 
 Bug fixes
 ~~~~~~~~~
+- Bug in :meth:`Series.concat`, Attribute Error raised with a None and Timestamp (:issue:`52093`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.0.1.rst:23]:
```suggestion
- Bug in :meth:`pd.concat` where an ``AttributeError`` raised with a None and Timestamp (:issue:`52093`)
```

Comment (mroeschke) [reply doc/source/whatsnew/v2.0.1.rst:23]: Also sorry could you move this to `v2.1.0.rst` instead?

File: doc/source/whatsnew/v2.0.1.rst [hunk 5]
```diff
@@ -20,10 +20,12 @@ Fixed regressions
...
 - Fixed bug in :func:`merge` when merging with ``ArrowDtype`` one one and a NumPy dtype on the other side (:issue:`52406`)
 - Bug in :meth:`Series.describe` not returning :class:`ArrowDtype` with ``pyarrow.float64`` type with numeric data (:issue:`52427`)
 - Fixed segfault in :meth:`Series.to_numpy` with ``null[pyarrow]`` dtype (:issue:`52443`)
 
+
```
Comment (mroeschke) [inline doc/source/whatsnew/v2.0.1.rst:28]: Can you undo this?

Comment (ramvikrams) [reply doc/source/whatsnew/v2.0.1.rst]: done

File: doc/source/whatsnew/v2.0.1.rst (hunk 4 above)
Comment (ramvikrams) [reply doc/source/whatsnew/v2.0.1.rst]: done

File: pandas/tests/internals/test_internals.py (hunk 2 above)
Comment (ramvikrams) [reply pandas/tests/internals/test_internals.py]: done

File: pandas/tests/internals/test_internals.py (hunk 1 above)
Comment (ramvikrams) [reply pandas/tests/internals/test_internals.py]: done

File: pandas/tests/reshape/concat/test_concat.py [hunk 6]
```diff
@@ -802,3 +802,14 @@ def test_concat_mismatched_keys_length():
...
+
+def test_concat_none_to_dttz():
+    # GH-52093
+    data = [{"A": None}]
+    df = DataFrame(data)
+    data_2 = [{"A": [pd.to_datetime("1990-12-20 00:00:00+00:00")]}]
+    df_2 = DataFrame(data_2)
+    result = concat([df, df_2])
+    expected = DataFrame({"A": [pd.to_datetime("1990-12-20 00:00:00+00:00")]})
+    tm.assert_frame_equal(result, expected)
```
Comment (mroeschke) [inline pandas/tests/reshape/concat/test_concat.py:815]: Looks like this test is failing from the logs

Comment (ramvikrams) [reply pandas/tests/reshape/concat/test_concat.py:815]: Actually  can you please suggest  a possible fix

Comment (mroeschke) [reply pandas/tests/reshape/concat/test_concat.py:815]: Check the `result` and make sure it has the same elements as `expected`

**Pull request closed without merging**